include/spdk/nvme.h

```diff
@@ -1161,6 +1161,23 @@ int spdk_nvme_ns_cmd_reservation_report(struct spdk_nvme_ns *ns,
 					void *payload, uint32_t len,
 					spdk_nvme_cmd_cb cb_fn, void *cb_arg);
 
+/**
+ * \brief Sends an abort command to the NVMe device for the request indicated
+ *  by cid
+ *
+ * \param ctrlr controller to the Nvme device
+ * \param qpair I/O queue pair associated with the request
+ * \param cid identification number of the command to be aborted.
+ * \param cb_fn a function pointer to the abort call back function
+ * \param cb_arg user argument to the abort command callback function
+ *
+ * \return 0 if successfully submitted, Less than 0 if it fails
+ */
```

Comment:
This is trickier than it seems because abort is an ADMIN command and is sent on the ADMIN queue. Therefore, to actually see cb_fn called you have to poll the admin queue - not the qpair you are passing in. That's really confusing, so at a minimum it should be well documented here.